GitCommands/Patches/PatchProcessor.cs

```diff
@@ -71,71 +88,54 @@ public static IEnumerable<Patch> CreatePatchesFromString(string patchText, Lazy<
-
+            string rawHeader = lines[lineIndex];
+            Match contentMatch = EscapeSequenceRegex().Match(rawHeader);
+            ReadOnlySpan<char> header = contentMatch.Groups["line"].ValueSpan;
+#if DEBUG
+            DebugHelpers.Assert(!CheckAnyEscapeSequenceRegex().IsMatch(header), "Git unexpectedly emits escape sequences in header other than at start/end.");
+#endif
+            Match headerMatch = PatchHeaderFileNameRegex().Match(header.ToString());
             if (!headerMatch.Success)
             {
                 return null;
             }
 
-            header = GitModule.ReEncodeFileNameFromLossless(header);
-
-            PatchProcessorState state = PatchProcessorState.InHeader;
-
-            string? fileNameA, fileNameB;
-
+            header = GitModule.ReEncodeFileNameFromLossless(header.ToString());
             bool isCombinedDiff = headerMatch.Groups["type"].Value != "git";
-
-            if (!isCombinedDiff)
+            if (!headerMatch.Success || (!isCombinedDiff && !headerMatch.Groups["filenameb"].Success))
             {
-                // diff --git a/GitCommands/CommitInformationTest.cs b/GitCommands/CommitInformationTest.cs
-                // diff --git b/Benchmarks/App.config a/Benchmarks/App.config
-                Match match = DiffCommandRegex().Match(header);
-
-                if (!match.Success)
-                {
-                    throw new FormatException("Invalid patch header: " + header);
-                }
-
-                fileNameA = match.Groups["filenamea"].Value.Trim();
-                fileNameB = match.Groups["filenameb"].Value.Trim();
+                throw new FormatException($"Invalid patch header: {header}");
             }
-            else
-            {
-                Match match = CombinedDiffCommandRegex().Match(header);
-
-                if (!match.Success)
-                {
-                    throw new FormatException("Invalid patch header: " + header);
-                }
 
-                fileNameA = match.Groups["filenamea"].Value.Trim();
-                fileNameB = null;
-            }
+            string fileNameA = headerMatch.Groups["filenamea"].Value.Trim();
+            string? fileNameB = isCombinedDiff ? null : headerMatch.Groups["filenameb"].Value.Trim();
 
-            string? index = null;
-            PatchChangeType changeType = PatchChangeType.ChangeFile;
-            PatchFileType fileType = PatchFileType.Text;
             StringBuilder patchText = new();
 
-            patchText.Append(header);
+            patchText.Append(ReaddEscapes(rawHeader, header, contentMatch));
             if (lineIndex < lines.Length - 1)
             {
                 patchText.Append('\n');
             }
 
+            PatchProcessorState state = PatchProcessorState.InHeader;
+            PatchChangeType changeType = PatchChangeType.ChangeFile;
+            PatchFileType fileType = PatchFileType.Text;
+            ReadOnlySpan<char> index = null;
```

Comment:
```suggestion
            ReadOnlySpan<char>? index = null;
```

Reply:
not a type

Reply:
So it is kind of a C# syntax / compiler issue!

Reply:
![image](https://github.com/gitextensions/gitextensions/assets/6248932/5d4daeb0-46a8-4bc6-b774-fd4b28919f99)

but the DEBUG assert below was incorrect, updating...